### DISCUSSION TITLE
S3 presigned POST URLs

### DISCUSSION THREAD

**Discussion by janczarn** [awslabs/aws-sdk-rust#711] [Q&A] [answered]
Hello,

In Javascript I am able to create presigned POST URLs for uploading to S3 using [this package](https://www.npmjs.com/package/@aws-sdk/s3-presigned-post). Presigned POST URLs allow a [POST policy](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-HTTPPOSTConstructPolicy.html) to be set which can include conditions such as `content-length-range`.

From what I can find, however, the Rust SDK allows [presigning PUT requests](https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/client/fluent_builders/struct.PutObject.html#method.presigned), but not POST requests. Have I missed something, or is this not possible with the Rust SDK yet?

Thanks!

**Comment by Velfi** [comment, marked as answer]
We don't have a crate for creating presigned `POST` requests. We'd consider this functionality to be a "high-level library" (HLL.)

We are not currently planning to implement any HLLs (e.g. the S3 transfer manager) until some time after our 1.0 release.
